src/library.tests/WorkloadInformationProviderTests.cs

```diff
@@ -47,63 +47,97 @@ public async void GetReachableServicesAsync_HeadlessService(int numServices, int
             
         }
 
-        private void ConfigureHeadlessService(int numServices, Func<int, string> namingFunction, int numAddresses, Func<int, string> addressHostNamingFunction)
-        {
-            var serviceList = new List<V1Service>();
-            // introducing this variable so we can add an endpoint with empty subset to have crash coverage
-            bool addSubset = false;
-            for (int i = 0; i < numServices; i++)
+        [Theory]
+        [InlineData(1, 100)]
+        [InlineData(5, 20)]
+        [InlineData(10, 3)]
+        [InlineData(1, 3)]
+        public async void GetReachableServicesAsync_PortsToIgnore_HeadlessService(int numServices, int numAddresses)
+        { 
+            List<int> portsToIgnore = new List<int> { 190, 570 };
```

Comment:
I believe this test right now is only testing the scenario where none of hte user ports match what the user configured in bridgetokubernetes/ignore-ports. For test to exercise all scenario you need to add these two ports or at least one of them to the input set (aka subsets list in that function). When ConfigureHeadlesService function runs, we need to make sure that on the response we are mocking for GetEndpointInNamespaceAsync there is some endpoint with the port that the user would want to ignore.